specs/CDPSessionIdSupport.md

```diff
@@ -0,0 +1,302 @@
+sessionId support for DevToolsProtocol method call and event
+===
+
+# Background
+A web page can have multiple DevToolsProtocol targets. Besides the default target for the top page, there are separate targets for iframes from different origin and web workers.
+
+The underlying DevToolsProtocol supports interaction with other targets by attaching to them and then using sessionId of the attachment in DevToolsProtocol method call message.
+The received DevToolsProtocol event messages also have sessionId field to indicate which target the event comes from.
+See [DevToolsProtocol Target domain](https://chromedevtools.github.io/devtools-protocol/tot/Target/) for more details.
+
+However, the current WebView2 DevToolsProtocol APIs like [CallDevToolsProtocolMethod](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2#calldevtoolsprotocolmethod)
+and [DevToolsProtocolEventReceivedEvent](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2devtoolsprotocoleventreceivedeventargs)
+doesn't support sessionId.
+
+To support interaction with different parts of the page, we are adding support for specifying sessionId for DevToolsProtocol method call API
+and retrieving sessionId for received DevToolsProtocol events.
+  
+# Conceptual pages (How To)
+
+To interact with different targets on the page, call DevToolsProtocol method `Target.setAutoAttach` with `flatten` set as `true`,
```

Comment:
This seems like a lot of work. Does every end dev who wants to use sessions have to do all of this? Is there some way we can make this easier for them?

Reply:
Rewrote the session, hopefully that makes easier to understand. The complexity comes from understanding CDP. I don't think that we should provide some easy wrapper for session/target management, at least without strong signal from developers.